dev/source-release.sh

```diff
@@ -66,7 +66,7 @@ tarball=$tag.tar.gz
 
 # be conservative and use the release hash, even though git produces the same
 # archive (identical hashes) using the scm tag
```

Comment:
I don't think that this comment is really necessary now. We are using a release hash still, but we're also using HEAD. However, the results are correct and this isn't any different than the proposed working solution of using `git ls-tree --name-only -r HEAD`.

Please let me know if the above comment can be removed, as to me it no longer makes sense. We also have less need to be conservative as we're using HEAD, and the script even issues and pushes a commit with the `version.txt` file, so HEAD is absolutely the correct place to be working from in this case.